### PR TITLE
Update resource with tag

### LINKED ISSUE #4398: Allow to tag a resource during updates
**Motivation**
It is now possible to tag a resource during creation (see #4301).

Users have expressed the need to extend the ability to optionally tag during updates

**Acceptance criteria**
1. When users update a resource, they can optionally provide a tag to tag it with the same event

### PR DESCRIPTION
**Pull request by dantb**
Fixes #4398

### CODE REVIEW

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/model/ResourceEvent.scala [hunk 1]
```diff
@@ -81,6 +81,8 @@ object ResourceEvent {
     *   the instant when this event was created
     * @param subject
     *   the subject which created this event
+    * @param tag
+    *   an optional user-specified tag attached to the resource on creation
```
Comment (dantb) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/model/ResourceEvent.scala:85]:
```suggestion
    *   an optional user-specified tag attached to the first revision on this resource
```

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/model/ResourceEvent.scala [hunk 2]
```diff
@@ -126,6 +128,8 @@ object ResourceEvent {
     *   the instant when this event was created
     * @param subject
     *   the subject which created this event
+    * @param tag
+    *   an optional user-specified tag attached to the resource on update
```
Comment (dantb) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/model/ResourceEvent.scala:132]:
```suggestion
    *   an optional user-specified tag attached to the current resource revision when updated
```

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/ResourcesImpl.scala [hunk 3]
```diff
@@ -86,12 +87,11 @@ final class ResourcesImpl private (
...
-      iri            <- expandIri(id, projectContext).toCatsIO
-      schemaRef      <- expandResourceRef(schema, projectContext)
-      resource       <- log.stateOr(projectRef, iri, ResourceNotFound(iri, projectRef)).toCatsIO
-      res            <- if (schemaRef.iri == resource.schema.iri) fetch(id, projectRef, schema.some)
-                        else eval(UpdateResourceSchema(iri, projectRef, schemaRef, resource.expanded, resource.rev, caller))
+      (iri, projectContext) <- expandWithContext(fetchContext.onModify, projectRef, id)
+      schemaRef             <- expandResourceRef(schema, projectContext)
+      resource              <- log.stateOr(projectRef, iri, ResourceNotFound(iri, projectRef)).toCatsIO
+      res                   <- if (schemaRef.iri == resource.schema.iri) fetch(id, projectRef, schema.some)
+                               else eval(UpdateResourceSchema(iri, projectRef, schemaRef, resource.expanded, resource.rev, caller))
```
Comment (shinyhappydan) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/ResourcesImpl.scala:94]: small but nice change

File: delta/sdk/src/test/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/ResourcesImplSpec.scala [hunk 4]
```diff
@@ -332,60 +332,60 @@ class ResourcesImplSpec
...
       "reject if project does not exist" in {
         val projectRef = ProjectRef(org, Label.unsafe("other"))
 
-        resources.update(myId, projectRef, None, 2, source).rejectedWith[ProjectContextRejection]
+        resources.update(myId, projectRef, None, 2, source, None).rejectedWith[ProjectContextRejection]
       }
 
       "reject if project is deprecated" in {
-        resources.update(myId, projectDeprecated.ref, None, 2, source).rejectedWith[ProjectContextRejection]
+        resources.update(myId, projectDeprecated.ref, None, 2, source, None).rejectedWith[ProjectContextRejection]
```
Comment (shinyhappydan) [inline delta/sdk/src/test/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/ResourcesImplSpec.scala:388]: Feels like there should be at least one test here passing a tag?

**Review by shinyhappydan — APPROVED** [review]

File: delta/sdk/src/test/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/ResourcesImplSpec.scala (hunk 4 above)
Comment (olivergrabinski) [reply delta/sdk/src/test/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/ResourcesImplSpec.scala:401]: ☝️

**Pull request merged**